src/utils/provider.ts

```diff
@@ -1,39 +1,46 @@
-import { Wallet, ethers } from 'ethers';
+import { Web3Provider } from '@ethersproject/providers';
+import { ethers } from 'ethers';
 
-export const getSignature = async (data: string, provider?: any) => {
+export const getSignature = async (data: string, provider?: Web3Provider) => {
```

Comment:
likewise here if we make `provider` non-optional then we can drop the following null check

Reply:
Same, this is going to be out of scope for me here